homeassistant/components/motionmount/config_flow.py

```diff
@@ -146,16 +151,87 @@ async def async_step_zeroconf_confirm(
         if user_input is None:
             return self.async_show_form(
                 step_id="zeroconf_confirm",
-                description_placeholders={CONF_NAME: self.discovery_info[CONF_NAME]},
+                description_placeholders={CONF_NAME: self.connection_data[CONF_NAME]},
                 errors={},
             )
 
+        return self._create_or_update_entry()
+
+    async def async_step_reauth(
+        self, user_input: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle re-authentication."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+
+        if TYPE_CHECKING:
+            assert self.reauth_entry
+
+        self.connection_data.update(self.reauth_entry.data)
+        return await self.async_step_auth()
```

Comment:
Please be sure to have a read of https://developers.home-assistant.io/blog/2024/10/21/reauth-reconfigure-helpers, a bit changed since this PR was created